homeassistant/components/cloud/account_link.py

```diff
@@ -60,6 +65,27 @@ async def async_provide_implementation(
     return []
 
 
+def _async_create_task(hass: HomeAssistant, coro: Any) -> asyncio.Task[Any]:
+    """Create a token task that can be cancelled during shutdown."""
+    task = hass.async_create_task(coro)
+    _PENDING_TASKS.add(task)
+
+    @callback
+    def _remove_task(_: Any) -> None:
+        _PENDING_TASKS.discard(task)
```

Comment:
The task will be provided as the argument to this function. I am wondering why you chose to use this redefinition of the callback function for each task instead of using a generic callback function that could accept any task.

Reply:
Ah, I didn't notice what's the argument type. I'll fix this and verify if it still works.

Reply:
Removed this part of code.